### ISSUE TITLE
uninitialized constant Chef::Knife::Bootstrap (NameError)

### ISSUE THREAD

**Issue by mlcooper** [chef/knife-windows#162]
Upgraded to Chef client 11.18 on my windows machine today and get an error when I try to upload a cookbook to the Chef server.  Has anyone else seen this error?  Uninstalling knife-windows does not solve the error.

    $ knife cookbook upload <cookbook_name> 
    c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/knife-windows-0.8.2/lib/chef/knife/bootstrap_windows_ssh.rb:23:in `<class:Knife>': uninitialized constant Chef::Knife::Bootstrap (NameError) 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/knife-windows-0.8.2/lib/chef/knife/bootstrap_windows_ssh.rb:22:in `<class:Chef>' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/knife-windows-0.8.2/lib/chef/knife/bootstrap_windows_ssh.rb:21:in `<top(required)>' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife/core/subcommand_loader.rb:37:in`load' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife/core/subcommand_loader.rb:37:in`block in load_commands' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife/core/subcommand_loader.rb:37:in`each' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife/core/subcommand_loader.rb:37:in`load_commands' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife.rb:120:in `load_commands' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/knife.rb:168:in `run' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/lib/chef/application/knife.rb:139:in `run' 
    from c:/opscode/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.18.0-x86-mingw32/bin/knife:25:in `<top (required)>' 
    from c:/opscode/chef/bin/knife:23:in `load' 
    from c:/opscode/chef/bin/knife:23:in `<main>'

**Comment by btm** [comment]
You probably haven't uninstalled the right copy of knife-windows, e.g. inside of the omnibus package versus out, i.e. ```chef gem``` vs ```gem```.

Try knife-windows 0.8.3.rc.0 (gem install --pre knife-windows) which has a workaround. 0.8.3 should ship soon.

**Comment by skazzyy** [comment]
So this solved the problem for me also... installed 11.18 and knife-windows and it doesn't work.  Until doing gem install knife-windows --pre.  So is the problem simply that knife-windows 0.8.2 doesn't work with chef 11.18?

**Comment by btm** [comment]
I've looked a couple times and it doesn't look like the ```knife bootstrap windows ssh``` code in bootstrap_windows_ssh.rb ever 'required' knife/bootstrap. I currently believe something changed in Chef that affected the load order. The bootstrap_windows_*.rb files should required knife/bootstrap, so I added that in  b9b2a0e81356e384bf157de3a1b8340a591c91c2 as I was prepping a release for https://github.com/chef/knife-windows/issues/131.
